strawberryfields/api/result.py

```diff
@@ -107,6 +102,28 @@ def state(self):
             raise AttributeError("The state is undefined for a stateless computation.")
         return self._state
 
+    @staticmethod
+    def combine_samples(samples_list, mode_order):
+        """Combine samples measured at different times into one nested array.
+
+        Args:
+            samples_list (list[array, tensor]): the sample measurements in a list
+            mode_order (list[int]): the mode order for the measured modes
+
+        Returns:
+            array: the samples in raising mode order with shape ``(shots, measured_modes)``
+        """
+        ret = [[] for _ in samples_list[0]]
+        for i, samples in enumerate(samples_list):
+            for j, single_sample in enumerate(samples):
```

Comment:
(Small suggestion): how about the `single_sample` -> `samples_for_run` renaming or so? `single_sample` sounds like a single outcome on a single mode